cogs/vxtwitconverter/eventsCore.py

```diff
@@ -21,29 +86,52 @@ async def _twit_replacer(self, message: Message):
             )
             return
 
-        # skips if the message has no embeds
-        if not any(embed.video for embed in message.embeds):
+        # the actual code part
+        vx_twtter_urls = self._convert_to_vx_twitter_url(message.embeds)
+
+        # no changed urls detected
+        if not vx_twtter_urls:
             return
 
-        # the actual code part
-        vx_twit_links = [
-            result.replace("https://twitter.com", "https://vxtwitter.com")
-            for result in URLExtract().find_urls(message.content)
-            if "https://twitter.com" in result
+        # constructs the message and replies with a mention
+        ok = await message.reply(self._urls_to_string(vx_twtter_urls))
+
+        # Remove embeds from user message if reply is successful
+        if ok:
+            await message.edit(suppress=True)
+
+    async def _on_edit_twit_replacer(self, message_before: Message, message_after: Message):
+        # skips if the message is sent by any bot
+        if not self._valid(message_after):
+            return
+
+        if not await self.config.guild(message_after.guild).get_attr(KEY_ENABLED)():
+            self.logger.info(
```

Comment:
1. Oops, missed this logging `debug` earlier.
2. Should we also move this into your helper `_valid` since it's common?